packages/material-ui/src/StepContent/StepContent.js

```diff
@@ -111,7 +111,7 @@ StepContent.propTypes = {
   /**
    * Collapse component.
    */
-  TransitionComponent: PropTypes.func,
+  TransitionComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]),
```

Comment:
I have spotted the inconsistency with a forwardRef warning.

Reply:
Funny thing. I always assumed a component is a function or an object because in my mind components can be classes. But only the instances are objects. The actual newable thing is still a function. So I don't know if either someone made the same mistake as I did or actually already thought about `forwardRef` and `memo`.

I locally experimented with `react-is` and want to follow up on this with an actual component validator that uses `isValidElementType` to improve the docs and dev warnings e.g. if one uses default imports when the module only has named exports. Then `PropTypes.object` would let it pass but react would later warn on `createElement`.

Reply:
Using `react-is` is definitely an interesting path to improve the component prop-type 👍 .

Reply:
Yes, I believe somebody was facing the `forwardRef` issue some time ago.